lib/logster/base_store.rb

```diff
@@ -131,6 +131,18 @@ def rate_limited?(ip_address, perform: false, limit: 60)
       not_implemented
     end
 
+    # find all pattern groups; returns an array of Logster::Group
+    def find_pattern_groups(load_messages: true)
+    end
+
+    # saves an instance of Logster::Group
+    def save_pattern_group(group)
+    end
+
+    # removes the Logster::Group instance associated with the given pattern
+    def remove_pattern_group(pattern)
+    end
+
```

Comment:
Do we want to force children classes to implement those methods by throwing `not_implemented`?